jdbcSource/src/main/java/io/vantiq/extsrc/jdbcSource/JDBC.java

```diff
@@ -73,17 +101,30 @@ public void setupJDBC(String dbURL, String username, String password) throws Van
      * @throws VantiqSQLException
      */
     public HashMap[] processQuery(String sqlQuery) throws VantiqSQLException {
-        // Check that connection hasn't closed
-        diagnoseConnection();
-        
         HashMap[] rsArray = null;
-        try (Statement stmt = conn.createStatement();
-            ResultSet rs = stmt.executeQuery(sqlQuery)) {
-            rsArray = createMapFromResults(rs);           
-        } catch (SQLException e) {
-            // Handle errors for JDBC
-            reportSQLError(e);
-        } 
+
+        if (isAsync) {
+            try (Connection conn = ds.getConnection();
+                 Statement stmt = conn.createStatement();
+                 ResultSet rs = stmt.executeQuery(sqlQuery)) {
+                rsArray = createMapFromResults(rs);
+            } catch (SQLException e) {
+                // Handle errors for JDBC
+                reportSQLError(e);
+            }
+        } else {
```

Comment:
Would it be better to just always use the connection pool?  Something like set it to a _pool of 1_ (`maximumPoolSize=1`)if we aren't async? Would that simplify the code?

Or do we want to do that only after this has been used for a while?